src/homework8/part2/Book.java

```diff
@@ -0,0 +1,24 @@
+package homework8.part2;
+
+public class Book implements Printable {
+
+    static String name;
```

Comment:
static is not applicable in a such case. static field is common(has same value) for all objects. So, when you create several books all of them will have the same name